message_processor.go

```diff
@@ -201,14 +201,14 @@ func (p *messageProcessor) SendPublic(ctx context.Context, chatID string, data [
 
 // SendPublicRaw takes encoded data, encrypts it and sends through the wire.
 func (p *messageProcessor) SendPublicRaw(ctx context.Context, chatName string, data []byte) ([]byte, error) {
-	var newMessage whisper.NewMessage
+	var newMessage whispertypes.NewMessage
 
 	wrappedMessage, err := p.tryWrapMessageV1(data)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to wrap message")
 	}
 
-	newMessage = whisper.NewMessage{
+	newMessage = whispertypes.NewMessage{
```

Comment:
Do we need to have this message created at this level? TTL, PoW and PoWTime remain always the same, no? Just passing a payload would make this more transport agnostic.

Reply:
I agree, but my goal with this PR is to make a first pass at getting rid of Geth-specific types and still having something that works. Baby steps. Also, you shouldn't forget that just because that approach would work well in this particular function, there are other places that still need a Geth-agnostic `NewMessage` struct. E.g. in the [`EnvelopesMonitor` class](https://github.com/status-im/status-protocol-go/blob/feature%2Fstatus-nim/transport/whisper/envelopes.go#L64), which will work over either version of the node.

Reply:
Yeah, understood. Not really in scope of this PR. And I realize that while here it is easy to move, at other locations it will be more difficult to get transport agnostic.